src/Advection/adapt_advection_order.jl

```diff
@@ -0,0 +1,111 @@
+using Oceananigans.Grids: topology
+
+"""
+    adapt_advection_order(advection, grid::AbstractGrid)
```

Comment:
This could just be `validate_advection_scheme(scheme, grid)`. We often refer to this kind of thing as a "validation"

Reply:
I tend to think at a validation as a non-returning function